br/pkg/lightning/backend/external/writer.go

```diff
@@ -181,27 +187,39 @@ func (b *WriterBuilder) Build(
 	}
 	p := membuf.NewPool(membuf.WithBlockNum(0), membuf.WithBlockSize(b.blockSize))
 	ret := &Writer{
-		rc: &rangePropertiesCollector{
-			props:        make([]*rangeProperty, 0, 1024),
-			currProp:     &rangeProperty{},
-			propSizeDist: b.propSizeDist,
-			propKeysDist: b.propKeysDist,
-		},
 		memSizeLimit:   b.memSizeLimit,
 		store:          store,
-		kvBuffer:       p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit)),
 		currentSeq:     0,
+		asyncFlushCh:   make(chan *asyncFlushTask),
 		filenamePrefix: filenamePrefix,
 		keyAdapter:     keyAdapter,
 		writerID:       writerID,
-		kvStore:        nil,
 		onClose:        b.onClose,
 		closed:         false,
 		multiFileStats: make([]MultipleFilesStat, 1),
 		fileMinKeys:    make([]tidbkv.Key, 0, multiFileStatNum),
 		fileMaxKeys:    make([]tidbkv.Key, 0, multiFileStatNum),
 	}
+	ret.doubleKVBuffer[0] = p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit))
+	ret.doubleKVBuffer[1] = p.NewBuffer(membuf.WithMemoryLimit(b.memSizeLimit))
+	ret.doubleRC[0] = &rangePropertiesCollector{
+		props:        make([]*rangeProperty, 0, 1024),
+		currProp:     &rangeProperty{},
+		propSizeDist: b.propSizeDist,
+		propKeysDist: b.propKeysDist,
+	}
+	ret.doubleRC[1] = &rangePropertiesCollector{
+		props:        make([]*rangeProperty, 0, 1024),
+		currProp:     &rangeProperty{},
+		propSizeDist: b.propSizeDist,
+		propKeysDist: b.propKeysDist,
+	}
+	ret.kvBuffer = ret.doubleKVBuffer[0]
+	ret.rc = ret.doubleRC[0]
```

Comment:
Memory consumption for new implement should be verified

Reply:
Yes I think we can test the memory consumption in the real end-to-end tests